### PR TITLE
Settings page

### PR DESCRIPTION
**Pull request by olaokoroafor**
Description:
  - Added a settings page where the user can change their name, city, state and the privacy setting
  - The user can leave page without saving changes
  - User can save and log out
  - User can access logout from this page

### CODE REVIEW

File: FoodFriends/app/src/main/java/com/example/foodfriends/activities/SettingsActivity.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+import com.example.foodfriends.R;
+import com.example.foodfriends.observable_models.UserObservable;
+import com.parse.ParseUser;
+
+import java.time.temporal.TemporalAccessor;
+import java.util.Observable;
+import java.util.Observer;
+
+public class SettingsActivity extends AppCompatActivity implements Observer {
```
Comment (nammyfb) [inline FoodFriends/app/src/main/java/com/example/foodfriends/activities/SettingsActivity.java:22]: Why a settings activity? And not utilize a PreferenceFragment?

**Review by nammyfb — CHANGES_REQUESTED** [review]

File: FoodFriends/app/src/main/java/com/example/foodfriends/activities/SettingsActivity.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+            }
+        });
+        tvSave.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                user.setState(etState.getText().toString());
+                user.setCity(etCity.getText().toString());
+                user.setName(etName.getText().toString());
+                user.setPrivateAccount(switchPrivacy.isChecked());
+                user.save_user();
```
Comment (asaxena1312) [inline FoodFriends/app/src/main/java/com/example/foodfriends/activities/SettingsActivity.java:70]: in a later pull request pls update to camel case

File: FoodFriends/app/src/main/java/com/example/foodfriends/activities/SettingsActivity.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+     */
+    private void userLogOut() {
+        ParseUser.logOut();
+        Intent i = new Intent(this, LogInActivity.class);
+        i.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
+        startActivity(i);
+    }
+
+    @Override
+    public void update(Observable o, Object arg) {
```
Comment (asaxena1312) [inline FoodFriends/app/src/main/java/com/example/foodfriends/activities/SettingsActivity.java:88]: what does update do?

File: FoodFriends/app/src/main/java/com/example/foodfriends/fragments/ProfileFragment.java [hunk 4]
```diff
@@ -109,21 +112,23 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
...
+            //btnLogOut.setOnClickListener(this);
             ivFindFriends.setOnClickListener(this);
             ivFollow.setVisibility(View.GONE);
             ivLock.setVisibility(View.GONE);
             display_content = true;
         } else {
             ivAddPfp.setVisibility(View.GONE);
-            btnLogOut.setVisibility(View.GONE);
+            ivSettingsIcon.setVisibility(View.GONE);
+            //btnLogOut.setVisibility(View.GONE);
```
Comment (asaxena1312) [inline FoodFriends/app/src/main/java/com/example/foodfriends/fragments/ProfileFragment.java:131]: remove

File: FoodFriends/app/src/main/java/com/example/foodfriends/fragments/ProfileFragment.java [hunk 5]
```diff
@@ -204,8 +209,9 @@ public void onClick(View v) {
             case R.id.ivAddPfp:
                 addPfp();
                 break;
-            case R.id.btnLogOut:
-                user_log_out();
+            case R.id.ivSettingsIcon://R.id.btnLogOut:
```
Comment (asaxena1312) [inline FoodFriends/app/src/main/java/com/example/foodfriends/fragments/ProfileFragment.java:212]: removed by mistake?

File: FoodFriends/app/src/main/java/com/example/foodfriends/misc/RestaurantServer.java [hunk 6]
```diff
@@ -209,7 +209,7 @@ public void done(List<UserToGo> togos, ParseException e) {
     private void parseGeneral(String apiKey, RestaurantListener listener) {
         ParseQuery<Restaurant> query = ParseQuery.getQuery(Restaurant.class);
         if (user.getCoordinates() != null) {
-            query.whereWithinKilometers("location_coordinates", user.getCoordinates(), max_radius);
+            query.whereWithinKilometers("location_coordinates", user.getCoordinates(), max_radius/1000);
```
Comment (asaxena1312) [inline FoodFriends/app/src/main/java/com/example/foodfriends/misc/RestaurantServer.java:212]: camel case

**Review by asaxena1312 — CHANGES_REQUESTED** [review]

File: FoodFriends/app/src/main/java/com/example/foodfriends/activities/SettingsActivity.java (hunk 1 above)
Comment (olaokoroafor) [reply FoodFriends/app/src/main/java/com/example/foodfriends/activities/SettingsActivity.java:22]: I wanted it to take up the whole screen and not include navigation bar, so this was the simplest way to do it!

File: FoodFriends/app/src/main/java/com/example/foodfriends/activities/SettingsActivity.java (hunk 2 above)
Comment (olaokoroafor) [reply FoodFriends/app/src/main/java/com/example/foodfriends/activities/SettingsActivity.java:70]: updated!

File: FoodFriends/app/src/main/java/com/example/foodfriends/activities/SettingsActivity.java (hunk 3 above)
Comment (olaokoroafor) [reply FoodFriends/app/src/main/java/com/example/foodfriends/activities/SettingsActivity.java:88]: update rerenders certain UI elements when an observable object changes

File: FoodFriends/app/src/main/java/com/example/foodfriends/fragments/ProfileFragment.java (hunk 4 above)
Comment (olaokoroafor) [reply FoodFriends/app/src/main/java/com/example/foodfriends/fragments/ProfileFragment.java:131]: Done!

File: FoodFriends/app/src/main/java/com/example/foodfriends/fragments/ProfileFragment.java (hunk 5 above)
Comment (olaokoroafor) [reply FoodFriends/app/src/main/java/com/example/foodfriends/fragments/ProfileFragment.java:212]: On purpose! Moved this logic to the settings page.

**Review by nammyfb — APPROVED** [review]

**Pull request merged**